pyiron_workflow/atomistics_library/macronodes.py

```diff
@@ -10,77 +9,41 @@
 )
 
 
-@single_value_node("instance")
-def get_instance(instance):
-    return instance
+def atomistics_meta_macro(task_generator_node_class, macro_name) -> type[Macro]:
+    def generic_macro(wf: Macro) -> None:
+        wf.tasks = task_generator_node_class()
+        wf.structures = generate_structures(instance=wf.tasks)
+        wf.calc = calc_with_calculator(task_dict=wf.structures)
+        wf.fit = analyse_structures(instance=wf.tasks, output_dict=wf.calc)
+        inputs_map = {
+            # Dynamically expose _all_ task generator input directly on the macro
+            "tasks__" + s: s
+            for s in wf.tasks.inputs.labels
+        }
+        inputs_map["calc__calculator"] = "calculator"
+        wf.inputs_map = inputs_map
+        wf.outputs_map = {"fit__result_dict": "result_dict"}
```

Comment:
Can this part be done more automatically? I mean can the workflow / macro object `wf` return a list of all input nodes which are not connected and all output nodes which are not connected and then just connect these to the nodes of the macro? While I understand this is not the behaviour everybody wants, I could see myself using this kind of functionality for a wide range of macros.

Reply:
This is _exactly_ what's done, but by default the macro-level channel labels are "{node_name}__{channel_name}" to make sure nothing conflicts and to reduce ambiguity.

Reply:
(well, not "exactly" as I-I and O-O are not "connections", connections are only for I-O/O-I, but anyhow the links are created to every child IO that is otherwise unconnected).

Reply:
I've been thinking about this in the back of my head during the meeting, and what I could see being a good compromise is offering some sort of keyword argument like `flatten_io_labels` or similar that tries to do this flattening automatically. I would still be more comfortable if it only happens when the user asks for it, but I agree it should be more convenient to do so than manually adding the snippet I use here.

Reply:
I really like the idea of having a special keyword like `flatten_io_labels` that would simplify it a lot.

Reply:
Yeah, I'm really sold on the idea now too. Are you comfortable merging this with the bespoke implementation here? I'm not sure how fast I'll get to implementing it a a generic feature, so I would be happy to have this one-off implementation as an example of how to do it right away, then come back and replace this with the keyword once the generic version exists.